### ISSUE TITLE
Need for type casting?

### ISSUE THREAD

**Issue by wacky6** [webmachinelearning/webnn#284]
This came up during review for softmax implementation https://chromium-review.googlesource.com/c/chromium/src/+/3856752/comments/267c926c_b14af87f

Softmax will likely only support float point types: https://github.com/webmachinelearning/webnn/issues/283

Currently, the spec provides no way to convert between data types, this might make it hard to use softmax with certain computation graphs.

This could impact models doing integer computations (e.g. image recognition models that takes in RGB24 image represented by uint8), and returns softmax-ed predictions (e.g. one-hot encoding).

For example, the following graph:
```
   Input (uint8)
-> matmul (uint32)
-> softmax
```

Perhaps WebNN needs a type-casting function?

In ML frameworks:
- [tf.cast](https://www.tensorflow.org/api_docs/python/tf/cast)
- [torch.Tensor.to with dtype](https://pytorch.org/docs/stable/generated/torch.Tensor.to.html)
- [Android NDK](https://developer.android.com/ndk/reference/group/neural-networks#group___neural_networks_1ggaabbe492c60331b13038e39d4207940e0a0a1bde5b34668d90eec4cfa4944e241d)
- [ONNX cast](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Cast)

**Comment by fdwr** [comment]
Definitely a gap, and included in v2 ops for transformers: https://github.com/webmachinelearning/webnn/issues/375#issuecomment-1674224992

```javascript
partial interface MLGraphBuilder {
  MLOperand cast(MLOperand input, MLOperandType operandType);
}
```

**Comment by inexorabletash** [comment]
Now that cast() landed (PR #478) can this be closed?

**Comment by inexorabletash** [comment]
Closing this as complete.
